SIL.Core/Reporting/ExceptionHandler.cs

```diff
@@ -24,33 +24,46 @@ public abstract class ExceptionHandler
 		//ExceptionHandler from there through Reflection. Otherwise we will simply use a console
 		//exception handler
 		/// <summary>
-		/// Initialize the ExceptionHandler. By default, the exceptionhandler will be initialized with a ConsoleExceptionHandler
-		/// unless he entry assembly uses a dependency on SIL.Windows.Forms.dll. In that case we default to the WinFormsExceptionHandler
+		/// Initialize the ExceptionHandler. By default, the exception handler will be initialized with a ConsoleExceptionHandler
+		/// unless the entry assembly uses a dependency on SIL.Windows.Forms.dll. In that case we default to the WinFormsExceptionHandler
 		/// </summary>
+		[Obsolete("Use Init<T>() instead, e.g. Init<ConsoleExceptionHandler>() or Init<WinFormsExceptionHandler>()")]
 		public static void Init()
 		{
-			if (_singleton == null)
-			{
-				//If we can't find the WinFormsExceptionHandler we'll use the Console
-				_singleton = GetObjectFromSilWindowsForms<ExceptionHandler>() ?? new ConsoleExceptionHandler();
-			}
-			else { throw new InvalidOperationException("An ExceptionHandler has already been set."); }
+			if (_singleton != null)
+				throw new InvalidOperationException($"An ExceptionHandler (of type ${_singleton.GetType()}) has already been set.");
+
+			//If we can't find the WinFormsExceptionHandler we'll use the Console
+			_singleton = GetObjectFromSilWindowsForms<ExceptionHandler>() ?? new ConsoleExceptionHandler();
 		}
 
 		/// <summary>
-		/// Use this method if you want to use an exeption handler besides the default.
+		/// Initialize the ExceptionHandler.
+		/// </summary>
+		/// <typeparam name="T">Type of exception handler</typeparam>
+		/// <exception cref="InvalidOperationException">If ExceptionHandler.Init has been called before</exception>
+		public static void Init<T>() where T: ExceptionHandler, new()
```

Comment:
I'm not entirely sure I see the benefit of having this version of Init. The client could just as easily construct the handler itself and pass it in, I think. I believe the only real point of having the Init method with no parameters was to prevent breaking existing apps, so why not just deprecate the plain Init() method and encourage clients to pass one in explicitly? In the particular case of the HearThis problem we're trying to address, none of the three Init methods would work anyway because the assemblies were incompatible. (Of course, either the template version or the one with an explicit handler passed as a parameter would actually throw an exception, so it would be easier for the developer to identify the problem.)